### ISSUE TITLE
Implement a proxy capability similar to LiteLLM 

### ISSUE THREAD

**Issue by chanwit** [weave-ai/weave-ai#13]
LiteLLM is a good example of how we should handle many models in the system.

In case of LLMs in a K8s environment, we already have information of LLMs as `LanguageModel CRs`.
- a model is presented as Flux `OCIRepository`
- a served mode is presented as `LanguageModel`.

A `LanguageModelProxy` could be an auto-discovery mechanism that scans LangageModel CRs and their endpoints then registers them as the list of models with the proxy.

**Comment by ishaan-jaff** [comment]
@chanwit I'm the maintainer of LiteLLM, why did you decide to re implement litellm proxy, instead of just using the LiteLLM proxy? 

What's missing in our solution?

**Comment by chanwit** [comment]
Hi @ishaan-jaff 

Thank you for reaching out.

I'm searching for a way to perform dynamic model and endpoint discovery, as it's crucial for our ecosystem. It would be beneficial if LiteLLM allowed the dynamic reloading of the config file, for example. That would help a lot.

**Comment by ghost** [comment]
@chanwit what does "dynamic reloading of the config file" mean? 

If you could share code mockup, etc. that would help 

tracking this here - https://github.com/BerriAI/litellm/issues/964

Once i understand what you're looking for, I'll shoot to have a v1 over to you asap

**Comment by chanwit** [comment]
Hi @krrishdholakia 

Our environment uses Kubernetes, and LLM instances are served by the LM-controller. When the LM-controller starts a new LLM instance, we want it to be automatically registered with a proxy. To achieve this, we plan to write a Kubernetes informer that monitors the creation and deletion of LLM instances, captures their endpoint information, and sends it to register the endpoint with the proxy, in this case, LiteLLM.

For LiteLLM to support this dynamic endpoint discovery, it needs to implement endpoint registration and deregistration, such as `POST` and `DELETE` at `/endpoints`, allowing for on-the-fly alias and endpoint registration without restarting the proxy process. From a security standpoint, we would limit this endpoint to local access only, or possibly use a Unix socket.

Does this make sense to you?

**Comment by ghost** [comment]
Yes I believe so. I'll have an initial version available for feedback out today.
